### DISCUSSION TITLE
bluez-alsa for Raspberry Pi 3A+ ?

### DISCUSSION THREAD

**Discussion by seamusdemora** [Arkq/bluez-alsa#692] [Q&A] [answered]
I have been struggling off & on for years (yes - **years**) with Bluetooth on my Raspberry Pis - I've never been able to get it working in a reliable, steady fashion. I have always used the 'Lite' version of the RPi OS (mostly Debian) - and this is apparently part of the problem. The fine people at the RPi organization have seen fit to never publish a shred of documentation on how to use Bluetooth on the Lite systems. And I was never *motivated* enough to dig it out for myself (the docs are quite arcane!) 

However, I now find myself in a situation where I ***need*** the Bluetooth audio capabilities. I've had a running argument with the RPi group for a few days now over the lack of documentation, but I don't expect to win an argument with wealthy [skinflints!](https://www.merriam-webster.com/dictionary/skinflint).  And so, I set out yesterday to *discover* how to build a reliable low-cost Linux "sound system" (i.e. a couple of old JBL Flip 5 BT speakers and a $25 RPi 3A). I've found that most things on Linux work if you're willing to put in time & effort - I'm hoping the same is true for BT audio. 

I just installed a new 'bookworm' OS earlier today, and did a survey of what BT audio software RPi includes. Here's the list: 

* pi-bluetooth 		Raspberry Pi 3 bluetooth; Loads BCM43430A1 firmware on boot
* libbluetooth3		Library to use the BlueZ Linux Bluetooth stack
* bluez				BlueZ is the official Linux Bluetooth protocol stack
* bluez-firmware		Firmware; required for BT dongles based on Broadcom BCM203x and Raspberry Pi chipset.
* alsa-topology-conf	ALSA config files; contains ALSA topology configuration files used by libasound2 for specific audio hardware  
* alsa-ucm-conf		ALSA Use Case Manager config files; ALSA UCM config of audio i/o names & routing for specific audio hardware; used with the alsaucm tool
* alsa-utils			Utilities for configuring and using ALSA; incl alsactl, alsaucm, aplay, arecord, speaker-test & several others

I began searching for `bluetooth audio using bluez and alsa`, and your GitHub/website made the front page. Your site appears to be current and active - both good signs. I've reviewed a small bit of the documentation & that looks OK. I was hoping for something that didn't require compilation/building from source, but I'll not make any judgments now. 

What I'd like to know is if you think your software is a good fit with my hardware, and will meet my "requirements" (stable and reliable). I'll likely use something like `mpg123` as a player - only because I know how to set it up. Can't think of anything else at the moment, so I'll close & leave it here for an answer. 

Thanks for your time,
~S

**Comment by graham8** [comment, marked as answer]
I can recommend bluez-alsa for your use case. I haven't tried RPiOS Lite but the OS I'm familiar with - OSMC - is similar in that it's Debian based and doesn't use pulseaudio or pipewire, and bluez-alsa works just fine.  Compiling bluez-alsa isn't difficult but I'm used to these things.  Above all, the support here is second to none.

The only difficult bit could be if you want to use those JBLs as a stereo pair. But you would have that problem with any BT setup.

**Reply by seamusdemora** [reply]
Thanks @graham8 !   Just out of curiosity, can you explain in a few words what is the relationship between `bluez-alsa` (this project) and `bluez` and `alsa` - which (I assume) are separate projects?  Is it some sort of "code integration" that you've done? 

P.S. the stereo pair isn't likely to be a big issue in the near future. All I'm trying to play now is "rain soundtracks" to help someone sleep... I don't believe stereo would add much  :)

**Reply by graham8** [reply]
This is not my project, but you're right - there's bluez and ALSA which you will find in just about every linux distirbution and bluez-alsa makes a bridge between them so that applications using ALSA (which all media applications do) can send and receive audio via BT.

HTH

**Reply by seamusdemora** [reply]
Great - that what I thought. I've got a couple more questions, but I guess I should open under a new topic as this one has been "answered"?
